README.md

```diff
@@ -114,11 +114,123 @@ chains = Vector{Any}(undef, nchains)
+
+- HMC:
+  ```julia
+  # HMC Sampler
+  # step size, number of leapfrog steps 
+  ϵ, n_leapfrogs = 0.1, 0.25
+  hmc = HMC(ϵ, n_leapfrogs)
+  ```
+
+  Equivalent to:
+
+  ```julia
+  metric = DiagEuclideanMetric(D)
+  hamiltonian = Hamiltonian(metric, ℓπ, ForwardDiff)
+  initial_ϵ = find_good_stepsize(hamiltonian, initial_θ)
+  integrator = Leapfrog(initial_ϵ)
+  kernel = NUTS{MultinomialTS, GeneralisedNoUTurn}(integrator)
+  adaptor = StanHMCAdaptor(MassMatrixAdaptor(metric), StepSizeAdaptor(δ, integrator))
+  nuts = HMCSampler(kernel, metric, adaptor)
+  ```
+
+- NUTS:
+  ```julia
+  # NUTS Sampler
+  # adaptation steps, target acceptance probability,
+  n_adapt, δ = 1000, 0.8
+  nuts = NUTS(n_adapt, δ)
+  ```
+
+  Equivalent to:
+
+  ```julia
+  metric = DiagEuclideanMetric(D)
+  hamiltonian = Hamiltonian(metric, ℓπ, ForwardDiff)
+  initial_ϵ = find_good_stepsize(hamiltonian, initial_θ)
+  integrator = Leapfrog(initial_ϵ)
+  kernel = HMCKernel(Trajectory{EndPointTS}(integrator, FixedNSteps(n_leapfrog)))
+  adaptor = NoAdaptation()
+  hmc = HMCSampler(kernel, metric, adaptor)
+  ```
+
+
+- HMCDA:
+  ```julia
+  #HMCDA (dual averaging)
+  # adaptation steps, target acceptance probability, target trajectory length 
+  n_adapt, δ, λ = 1000, 0.8
+  hmcda = HMCDA(1000, 0.8, 1.0)
+  ```
+
+  Equivalent to:
+
+  ```julia
+  metric = DiagEuclideanMetric(D)
+  hamiltonian = Hamiltonian(metric, ℓπ, ForwardDiff)
+  initial_ϵ = find_good_stepsize(hamiltonian, initial_θ)
+  integrator = Leapfrog(initial_ϵ)
+  kernel = HMCKernel(Trajectory{EndPointTS}(integrator, FixedIntegrationTime(λ)))
+  adaptor = StanHMCAdaptor(MassMatrixAdaptor(metric), StepSizeAdaptor(δ, integrator))
+  hmcda = HMCSampler(kernel, metric, adaptor)
+  ```
+
+Moreover, there's some flexibility in how these samplers can be initialized. 
+For example, a user can initialize a NUTS (as well as HMC and HMCDA) sampler with their own metric and integrator. 
+This can be done as follows:
+  ```julia
+  nuts = NUTS(n_adapt, δ, metric = :diagonal) #metric = DiagEuclideanMetric(D) (Default!)
+  nuts = NUTS(n_adapt, δ, metric = :unit)     #metric = UnitEuclideanMetric(D)
+  nuts = NUTS(n_adapt, δ, metric = :dense)    #metric = DenseEuclideanMetric(D)
+  # Provide your own :AbstractMetric
+  metric = DiagEuclideanMetric(10)
+  nuts = NUTS(n_adapt, δ, metric = metric) 
+
+  nuts = NUTS(n_adapt, δ, integrator = :leapfrog)         #integrator = Leapfrog(ϵ) (Default!)
+  nuts = NUTS(n_adapt, δ, integrator = :jitteredleapfrog) #integrator = JitteredLeapfrog(ϵ, 0.1ϵ)
+  nuts = NUTS(n_adapt, δ, integrator = :temperedleapfrog) #integrator = TemperedLeapfrog(ϵ, 1.0)
+
+  # Provide your own :AbstractIntegrator
+  integrator = JitteredLeapfrog(ϵ, 0.2ϵ)
+  nuts = NUTS(n_adapt, δ, integrator = integrator) 
+  ```
+
+Finally, bare in mind that the convinience constructors return `AbstractMCMC.AbstractSampler` and therefore they must be used using the `AbstractMCMC` interface as in the case of `HMCSampler`. 
```

Comment:
```suggestion
```

Is this important to specify? Now there is no other interface, right? It's all the same `sample` method, no?